### PR TITLE
Add bulk downloading

### PR DESCRIPTION
**Pull request by gluap**
We should also offer bullk downloading if we offer all those other bulk actions.

### CODE REVIEW

File: api/obs/api/routes/tracks.py [hunk 1]
```diff
@@ -58,6 +59,39 @@ async def _return_tracks(req, extend_query, limit, offset, order_by=None):
...
+
+async def tar_of_tracks(req, files):
+
+    response = await req.respond(content_type="application/x-gtar", headers={'Content-Disposition': 'attachment; filename="tracks.tar.bz2"'})
+
+    helper = StreamerHelper(response)
+
+    tar = tarfile.open(name=None, fileobj=helper, mode="w|bz2", bufsize=256 * 512)
+    for fname in files:
+        logging.info(f"sending {fname}")
```
Comment (opatut) [inline api/obs/api/routes/tracks.py:87]:
```suggestion
        log.info("Write file to tar: %s", fname)
```

File: api/obs/api/routes/tracks.py [hunk 2]
```diff
@@ -58,6 +59,39 @@ async def _return_tracks(req, extend_query, limit, offset, order_by=None):
         },
     )
 
+class StreamerHelper:
```
Comment (opatut) [inline api/obs/api/routes/tracks.py:62]: This whole class and the tar_of_tracks function should maybe move to utils?

File: frontend/src/pages/MyTracksPage.tsx [hunk 3]
```diff
@@ -233,12 +235,16 @@ function TracksTable({ title }) {
   };
 
   const bulkAction = async (action: string) => {
-    await api.post("/tracks/bulk", {
+    const data = await api.post("/tracks/bulk", {
```
Comment (opatut) [inline frontend/src/pages/MyTracksPage.tsx:238]: Rename `data` to `response`, that's what the object is called.

**Review by opatut — COMMENTED** [review]
I'll test this soon. Found a few code things already. I'll change it myself when I get around to testing it.

**Comment by gluap** [comment]
I added the requested changes from the initial review.

**Comment by opatut** [comment]
I don't like how the paths in the file are all `data/tracks/$username/$file.csv`, can't we strip the prefix?

**Comment by gluap** [comment]
Actually I was entirely focused on getting the asynchronous streaming tar to work and yes - the directory structure wasn't pretty. I modified it to strip the common part of the path. So for a single user it strips everything including the user directory, but if used as an admin in the future, the user name directorys will be retained.

**Comment by opatut** [comment]
Thanks, I didn't know about `os.path.commonpath` and I love it.

**Pull request merged**